server/rpc_test.go

```diff
@@ -532,3 +534,59 @@ func defaultTicket(t *testing.T) *net.Ticket {
 		RecipientRandHash: pm.RandBytes(123),
 	}
 }
+
+func StubBroadcastSessionsManager() *BroadcastSessionsManager {
+	sess1 := &BroadcastSession{
+		Broadcaster: StubBroadcaster2(),
+		ManifestID:  core.RandomManifestID(),
+	}
+	sess2 := &BroadcastSession{
+		Broadcaster: StubBroadcaster2(),
+		ManifestID:  core.RandomManifestID(),
+	}
+	return &BroadcastSessionsManager{broadcastSessions: []*BroadcastSession{sess1, sess2}, sessLock: &sync.Mutex{}}
+}
+
+func TestSelectFromList_Successt(t *testing.T) {
+	bsm := StubBroadcastSessionsManager()
+	expectedSess := bsm.broadcastSessions[0]
+	sess := bsm.selectFromList()
+
+	assert := assert.New(t)
+	assert.Equal(expectedSess, sess)
+	assert.Len(bsm.broadcastSessions, 1)
+}
+
+func TestRemoveFromList_Successt(t *testing.T) {
+	bsm := StubBroadcastSessionsManager()
+	expectedSess := bsm.broadcastSessions[0]
+	remove := bsm.broadcastSessions[1]
+	bsm.removeFromList(remove)
+	glog.Error(remove)
+
+	assert := assert.New(t)
+	assert.Equal(expectedSess, bsm.broadcastSessions[0])
+	assert.Len(bsm.broadcastSessions, 1)
+}
+
+func TestAddToList_Success(t *testing.T) {
+	bsm := StubBroadcastSessionsManager()
+	sess := &BroadcastSession{
+		Broadcaster: StubBroadcaster2(),
+		ManifestID:  core.RandomManifestID(),
+	}
+
+	bsm.addToList(sess)
```

Comment:
Sanity check the "before" case as well, eg that the list length is as expected.

Also test concurrent executions  of `AddToList` and `SelectFromList` and run with the `-race` flag.

Reply:
Do you mean this? 
```golang
bsm := StubBroadcastSessionsManager()
expectedSess := bsm.broadcastSessions[1]

assert := assert.New(t)
assert.Len(bsm.broadcastSessions, 2)

sess := bsm.selectFromList()

assert.Equal(expectedSess, sess)
assert.Len(bsm.broadcastSessions, 1)
```

As for testing concurrent executions.... do you mean on my local environment?